frame/uniswap-v2/src/lib.rs

```diff
@@ -349,6 +387,106 @@ pub mod pallet {
 		}
 	}
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_n: T::BlockNumber) -> Weight {
+			let current_timestamp = T::Time::now();
+			Pools::<T>::iter_keys().fold(Weight::zero(), |weight, pool_id| {
```

Comment:
are pools permissionless? may be add to do than to make this off chain worker as it will grow large (i know some AMM has 3.4k pools)

Reply:
Good catch, probably nice to make it optional per pool :thinking: